src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -68,58 +68,52 @@ internal class SqlServerSearchService : SearchService
 
         protected override async Task<SearchResult> SearchInternalAsync(SearchOptions searchOptions, CancellationToken cancellationToken)
         {
-            using (var connection = new SqlConnection(_configuration.ConnectionString))
-            {
-                connection.Open();
+            SearchResult searchResult;
 
-                SearchResult searchResult;
+            // If we should include the total count of matching search results
+            if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+            {
+                searchResult = await SearchImpl(searchOptions, false, cancellationToken);
 
-                // If we should include the total count of matching search results
-                if (searchOptions.IncludeTotal == TotalType.Accurate && !searchOptions.CountOnly)
+                // If this is the first page and there aren't any more pages
+                if (searchOptions.ContinuationToken == null && searchResult.ContinuationToken == null)
                 {
-                    // Begin a transaction so we can perform two atomic reads.
-                    using (var transaction = connection.BeginTransaction(IsolationLevel.ReadCommitted))
```

Comment:
I removed this (for now) and updated the User Story [#AB70783](https://microsofthealth.visualstudio.com/Health/_workitems/edit/70783).